### PR TITLE
Added protocol option to wait_for 

### LINKED ISSUE #1674: Specify configuration file for supervisorctl
This is a feature request and partially workaround for a bug in Supervisord that concerns the "supervisorctl" module.

Currently, there is a [case when `supervisorctl` cannot be started without arguments](https://github.com/Supervisor/supervisor/pull/62). A [patch is on its way](https://github.com/Supervisor/supervisor/pull/95), but has not been merged for a long time. I propose a `conf=/etc/supervisor.conf` argument to the supervisoctl module. This will both work around the bug, as well as add another feature to the plugin.

### PR DESCRIPTION
**Pull request by dulin**
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

wait_for utility 

##### ANSIBLE VERSION

```
2.3.0
```

##### SUMMARY

Added protocol support for wait_for, that give the chance to check services such as dns, freeradius etc...

### CODE REVIEW

File: lib/ansible/modules/utilities/logic/wait_for.py [hunk 1]
```diff
@@ -369,13 +374,15 @@ def _create_connection(host, port, connect_timeout):
     Returns:
         Socket object
     """
-    if sys.version_info < (2, 6):
```
Comment (gregswift) [inline lib/ansible/modules/utilities/logic/wait_for.py:372]: I don't think we should be removing older python support just yet.

https://groups.google.com/forum/#!topic/ansible-devel/BvhGM3wjH54

@abadger or @bcoca ?

**Comment by nerzhul** [comment]
why not add ICMP & ICMPv6 too ?

File: lib/ansible/modules/utilities/logic/wait_for.py (hunk 1 above)
Comment (bcoca) [reply lib/ansible/modules/utilities/logic/wait_for.py:372]: Python 2.4/2.5 already do not work with Ansible >= 2.4

**Comment by gregswift** [comment]
@dulin if you rebase generally i'm good with state of things.  adding icmp wouldnt be a bad thing but isn't a requirement afaict

**Comment by dulin** [comment]
@gregswift Please close that request i found issue where i'm not able to sort. The problem is with udp protocol, my solution all the time shows as online. I was interested how nmap detects udp listening ports and possibly is sorted using libpcap and require root access:
`
nmap -sU -p 1812 172.30.255.20

You requested a scan type which requires root privileges.
QUITTING!
`
That should be sorted by other way. Using: /proc/net/udp can be done for test on remote host. 
For example:
If you have playbook where you implementing firewall on your server and you expecting udp port should be closed for your control host (checking policy) that will doesn't work, anyway ansible should be run by normal user.

**Comment by gregswift** [comment]
close_me

**Pull request closed without merging**